ruma-api/src/lib.rs

```diff
@@ -267,6 +267,26 @@ pub struct Metadata {
     pub requires_authentication: bool,
 }
 
+/// Generates a `match` statement for the given expression that returns early.
+#[macro_export]
+macro_rules! try_deserialize {
+    ($kind:ident, $call:expr $(,)*) => {
+        ::ruma_api::try_deserialize!(@ $kind, $kind, $call)
+    };
+    (@ request, $kind:ident, $call:expr) => {
```

Comment:
Nice, you made it work! :)

I think the way I've seen this written usually is like this: `@request`, so without the extra space. Where did you get the idea with the `@`?

Reply:
I remember reading somewhere that the `@` was meant to signal internal details of the macro, if you meant with spaces I probably added that as I can't remember exactly where I saw that.

I was surprised that doubling the token `$kind, $kind` actually worked. Did you have a specific technique in mind?

Reply:
> I remember reading somewhere that the `@` was meant to signal internal details of the macro

Okay :)
Yeah that's what I use too and would basically be my reasoning as well. So no, I didn't have anything else in mind, just the same thing without the extra space, like you've implemented now.

> I was surprised that doubling the token `$kind, $kind` actually worked.

Why not? :)